src/iris/api.py

```diff
@@ -6081,13 +6081,19 @@ def on_get(self, req, resp):
 
         # deduplicate messages with same destination
         unique_messages = []
-        unique_destinations = set()
+        unique_destinations = {}
         for message in messages:
-            if message.get("destination") and message.get("destination") not in unique_destinations:
-                unique_destinations.add(message["destination"])
+            if message.get("destination") and message.get("mode") not in unique_destinations.get(message["destination"], []):
+                if message["destination"] not in unique_destinations:
+                    unique_destinations[message["destination"]] = [message["mode"]]
+                else:
+                    unique_destinations[message["destination"]] = unique_destinations[message["destination"]].append(message["mode"])
```

Comment:
We could avoid a bit of boilerplate by using a defaultdict(set) instead of a dict.

Eg

```python
unique_destinations = defaultdict(set)
for message in messages:
  if message.get("destination") and message.get("mode") not in unique_destinations[message["destination"]]:
     unique_destinations[message["destination"]].add(message["mode"])
# and so on
```